module-system/sov-modules-api/src/pub_key_hex.rs

```diff
@@ -80,6 +80,14 @@ impl TryFrom<&PublicKeyHex> for DefaultPublicKey {
     }
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for PublicKeyHex {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
```

Comment:
Since this wasn't introduced on this PR, I'll leave it as optional.

Shouldn't we generate a secret key, then public, then finally its string representation? If we end with random strings, they will almost never encode to a key that can be used

Reply:
Agree. 

And if we want to test "ArbitraryString" as PublicKeyHex, we will do exactly that: generate string manually and contruct PubliKeyHex from it